src/tools/brush/ScissorsTool.js

```diff
@@ -0,0 +1,329 @@
+   * @param {HTMLElement} element
+   * @param {string} strategy The strategy to be used on Tool
+   * @private
+   * @returns {void}
+   */
+  _changeCursor(element, strategy) {
+    // Necessary to avoid setToolCursor without elements, what throws an error
+    if (!this.element) {
+      return;
+    }
+
+    const cursorList = {
+      FILL_INSIDE: scissorsFillInsideCursor,
+      FILL_OUTSIDE: scissorsFillOutsideCursor,
+      ERASE_OUTSIDE: scissorsEraseOutsideCursor,
+      ERASE_INSIDE: scissorsEraseInsideCursor,
+      default: scissorsFillInsideCursor,
+    };
+
+    const newCursor = cursorList[strategy] || cursorList.default;
+
+    setToolCursor(element, newCursor);
+    external.cornerstone.updateImage(element);
+  }
+
+  /**
+   * Event handler for MOUSE_UP/TOUCH_END during handle drag event loop.
+   *
+   * @private
+   * @method _applyStrategy
+   * @param {(CornerstoneTools.event#MOUSE_UP|CornerstoneTools.event#TOUCH_END)} evt Interaction event emitted by an enabledElement
+   * @returns {void}
+   */
+  _applyStrategy(evt) {
+    this._applySegmentationChanges(evt);
+    this._resetHandles();
+    external.cornerstone.updateImage(evt.detail.element);
+  }
+
+  /**
+   * Sets the start and end handle points to empty objects
+   *
+   * @private
+   * @method _resetHandles
+   * @returns {undefined}
+   */
+  _resetHandles() {
+    this.handles = {
+      points: [],
+    };
+
+    this.currentHandle = 0;
+  }
+
+  /**
+   * Adds a point on mouse click in polygon mode.
+   *
+   * @private
+   * @param {Object} eventData - data object associated with an event.
+   * @returns {undefined}
+   */
+  _addPoint(eventData) {
+    // If this is not the first handle
+    if (this.handles.points.length) {
+      // Add the line from the current handle to the new handle
+      this.handles.points[this.currentHandle - 1].lines.push({
+        x: eventData.currentPoints.image.x,
+        y: eventData.currentPoints.image.y,
+        lines: [],
+      });
+    }
+
+    // Add the new handle
+    this.handles.points.push({
+      x: eventData.currentPoints.image.x,
+      y: eventData.currentPoints.image.y,
+      lines: [],
+    });
+
+    // Increment the current handle value
+    this.currentHandle += 1;
+
+    // Force onImageRendered to fire
+    external.cornerstone.updateImage(eventData.element);
+  }
+
+  /**
+   * Change Strategy Method
+   * @param { string } strategy
+   * @private
+   * @returns {void}
+   */
+  _changeStrategy(strategy = 'default') {
+    this.activeStrategy = strategy;
+    this._changeCursor(this.element, strategy);
+    this._resetHandles();
+  }
+
+  _applySegmentationChanges(evt) {
+    const points = this.handles.points;
+    const { image, element } = evt.detail;
+
+    const {
+      labelmap3D,
+      currentImageIdIndex,
+      activeLabelmapIndex,
+    } = getters.getAndCacheLabelmap2D(element);
+
+    const toolData = getters.labelmapBuffers(element, activeLabelmapIndex);
```

Comment:
The brush store does all this for you, you can replace lines 286 - 289 with:

```js
const segmentationData = labelmap3D.labelmaps2D[currentImageIdIndex].pixelData;
```

:)

An aside @dannyrb: I've used the `pixelData` property for all the labelmap2D stuff as I just copied DICOM nomenclature. However maybe its a bit hard-and-loose to call our labelmaps `pixelData`, and `segmentationData` is a better name?

Reply:
I guess that @dannyrb didn't have a chance to dig it, any way, we can change this name later in another PR and close this one. I'll resolve it for now @JamesAPetts , feel free to open it again if needed.